compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -702,9 +703,11 @@ class ProtobufGenerator(
       )
     }
     val maybeUnknownFields =
-      if (message.preservesUnknownFields)
+      if (message.preservesUnknownFields && message.unknownFieldsAnnotations.nonEmpty) {
+        Seq(ConstructorField.unknownFields(message.unknownFieldsAnnotations))
```

Comment:
Is long as `preservesUnknownFields` is true, can't we always call the new `ConstructorField.unknownFeilds(...)` and possibly pass an empty list? This would be preferable IMO over branching.